src/util.ts

```diff
@@ -427,3 +427,14 @@ export function utcToLocal(input : string, format = SQL_DATETIME_FORMAT) {
 export function localToUTC(input : string, format = SQL_DATETIME_FORMAT) {
     return dayjs(input).utc().format(format);
 }
+
+export function hostNameRegexPattern(mqtt = false) {
```

Comment:
util.ts have to compile to util.js.

If this is frontend only, you can use `util-frontend.js` instead.

Also, I think I need a auto test for this function to ensure it is working fine. As ipRegexPattern and hostNameRegexPattern is not identical as original.

Reply:
Do you mean unit test or e2e? I can't seem to find any unit tests that cover this kind of function but I'll look into adding a few e2e tests

Reply:
A unit test will be enough. Maybe I will setup a basic case for you.

Reply:
I agree, unit tests would be more appropriate. If you don't mind setting it up hat would be great thanks.

Reply:
I almost forget. There is no frontend unit test, because I have no idea how to add jest for frontend code. Since it is using `import/export` rather than commonjs `require(...)`

> SyntaxError: Cannot use import statement outside a module

May need some time to dig into the problem or looking for another unit test tool.

Reply:
I can look into add jest if you like?

Reply:
Yes, you can take a look of old code:
https://github.com/louislam/uptime-kuma/tree/1.17.0

package.json > jest-frontend

Reply:
I submitted a seperate issue for this as it can be implemented independently from this issue.